### PR TITLE
Adding VPC support for construct aws-lambda-elasticsearch-kibana

### PR DESCRIPTION
**Pull request by joyeezhao**
*Description of changes:*

Adding VPC support for construct aws-lambda-elasticsearch-kibana.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: df517956470250689d8732e52724a0b5c63f89e5
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f7f816b7-bf93-4943-b375-80a3c67d661c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,9 @@
 .vscode/
 /.favorites.json
 
+# WebStorm extension
```
Comment (biffgaut) [inline .gitignore:5]: What is this?

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/lib/index.ts [hunk 2]
```diff
@@ -89,9 +105,32 @@ export class LambdaToElasticSearchAndKibana extends Construct {
...
+      throw new Error("Error - More than 1 VPC specified for Lambda function");
+    }
+
+    if ((props.deployVpc || props.vpcProps || props.existingVpc) &&
+        props.esDomainProps && props.esDomainProps.vpcOptions) {
+      throw new Error("Error - More than 1 VPC specified in the properties");
+    }
+
+    if (props.deployVpc || props.existingVpc) {
+      this.vpc = defaults.buildVpc(scope, {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/lib/index.ts:119]: Missing this check here:

```
  if (props.deployVpc && props.existingVpc) {
          throw new Error("More than 1 VPC specified in the properties");
        }
```

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.deployFunctionWithVpc.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      subnetType: ec2.SubnetType.PUBLIC,
+    }
+  ]
+};
+
+new LambdaToElasticSearchAndKibana(stack, 'test-lambda-elasticsearch-kibana3', {
+  lambdaFunctionProps: lambdaProps,
+  domainName: "deploywithvpctest",
+  deployVpc: true,
+  vpcProps
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.deployFunctionWithVpc.ts:56]: How about a test that provides deployVpc and no props (an accepted scenario). The construct should create a default VPC that is appropriate for its use.

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (joyeezhao) [reply .gitignore:5]: Will delete this, it should be my IDE auto added line.

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/lib/index.ts (hunk 2 above)
Comment (joyeezhao) [reply source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/lib/index.ts:119]: This check is covered in `defaults.CheckProps(props)`, should add one more check here?

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.deployFunctionWithVpc.ts (hunk 3 above)
Comment (joyeezhao) [reply source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.deployFunctionWithVpc.ts:56]: Currently in index.ts, the vpc is only added on lambda(ElasticSearch cluster remain staying in public), so to successfully call ElasticSearch cluster, the vpc needed to be configured (using vpcProps) to make the call from lambda. If user provides deployVpc with no props and still want to make a successful call, we may need to override esDomainProps to add vpc to ElasticSearch cluster as well. Is that the next step I should go with?

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: f825bd06e01eef13b66b268e09c1ba664bf0228e
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7b13bcfa-a4dc-417b-ba3a-c5ec9cf6f2ea%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: df6b0a85f1f8fd4ce3316397d9d0d6a3d64e7294
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0c4de4e0-50fd-4a00-95b4-b3b683f7517b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: e2e88f99749cc90b74fc44825176eec5b3cf0cd6
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e7f732c5-06bf-4c87-94c7-320d1d2706c2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: cd2cff155f659becc8eb0f5dba3a475cb337bc68
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cc503d99-50f9-4d0d-9f3a-9f70ae41daf9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d0980a347b51273b38503c7e2b23e326581a5da7
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3208c818-46a2-4560-9e3b-a414c5a42802%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: fd072e3beb2dbbe5179a5968db3264a8394133dd
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7ce75bcb-6209-4daf-9f5f-fa6498048092%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 4f3efc3269c164f8760adeaecbea4745bbebde6f
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=27ebe664-3470-45b7-ab73-9ef5ef785c9c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: c24c1ccb608a08911cc167457f5ba4267ac9cd11
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=84849c5e-7856-4d8d-8076-7a2c2cca9b38%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 4f3efc3269c164f8760adeaecbea4745bbebde6f
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4083e7dd-76d2-4734-b792-6736e77529e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: c24c1ccb608a08911cc167457f5ba4267ac9cd11
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=098a24c8-ce40-4a29-90aa-24a186bab854%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — CHANGES_REQUESTED** [review]
Couple of questions and small changes requested.

One key question we will ask before merging is "Have you launched this construct with a Lambda function that accesses the ES cluster and proven that, in addition to launching successfully, creates an architecture that functions correctly?"

You can do this however you wish - an easy way is to grab the template for a deleted integration test stack out of CloudFormation (the Template tab on the right) and launch that template.

**Pull request closed without merging**